homeassistant/components/water_heater/__init__.py

```diff
@@ -359,6 +360,35 @@ async def async_set_operation_mode(self, operation_mode: str) -> None:
         """Set new target operation mode."""
         await self.hass.async_add_executor_job(self.set_operation_mode, operation_mode)
 
+    async def async_handle_set_operation_mode(self, operation_mode: str) -> None:
```

Comment:
If we want to have this method in the `WaterHeaterEntity` class it should be marked with the `final` decorator to prevent integrations from overriding it. Another option would be to move it outside the class like done here: https://github.com/emontnemery/home-assistant/blob/51c23a1ea5c31708b3264426b3b7be176d2d12ee/homeassistant/components/light/__init__.py#L630-L639